core/src/replay_stage.rs

```diff
@@ -758,20 +760,52 @@ impl ReplayStage {
         result
     }
 
-    fn verify_and_process_entries(
+    fn verify_entries(
         bank: &Arc<Bank>,
         entries: &[Entry],
-        shred_index: usize,
         bank_progress: &mut ForkProgress,
-    ) -> Result<()> {
+    ) -> std::result::Result<(), BlockError> {
+        let hashes_per_tick = bank.hashes_per_tick().unwrap_or(0);
+        if !entries.verify_tick_hash_count(&mut bank_progress.tick_hash_count, hashes_per_tick) {
+            return Err(BlockError::InvalidTickHashCount);
+        }
+
+        let next_bank_tick_height = bank.tick_height() + entries.tick_count();
+        let max_bank_tick_height = bank.max_tick_height();
+        if next_bank_tick_height > max_bank_tick_height {
+            return Err(BlockError::InvalidTickCount);
+        } else if next_bank_tick_height == max_bank_tick_height {
+            if let Some(last_entry) = entries.last() {
```

Comment:
I think we should just early exit this function if entries.is_empty(), and then we can just hard unwrap here.